File: Subset Sum Problem.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

bool subsetSum( int arr[], int n, int s)
{
    if(n==-1)
        return s==0;
    return subsetSum(arr,n-1,s-arr[n]) || subsetSum(arr,n-1,s);
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */  
    int t;
    cin>>t;
    while(t--){
        int n,s;
        cin>>n>>s;
        int arr[n];
        for(int i=0;i<n;i++){
            cin>>arr[i];
        }
        if(subsetSum(arr,n-1,s))
            cout<<"YES"<<endl;
        else cout<<"NO"<<endl;
    }
    return 0;
}

File: Subset Sum Problem.py
def findSubsetSum(arr, k):
    dp = [[False for i in range(k+1)] for i in range(len(arr)+1) ]
    for i in range(len(arr)+1):
        dp[i][0] = True
    for i in range(1, len(arr)+1):
        for j in range(1, k+1):
            if j< arr[i-1]:
                dp[i][j] = dp[i-1][j]
            else:
                dp[i][j] = dp[i-1][j] or dp[i-1][j-arr[i-1]]
    # print(dp)
    return dp[len(arr)][k]


for _ in range(int(input())):
    n, k = list(map(int, input().split()))
    arr = list(map(int, input().split()))
    print(*[ "YES" if findSubsetSum(arr, k) else "NO"])
